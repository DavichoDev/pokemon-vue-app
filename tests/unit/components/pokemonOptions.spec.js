import { shallowMount } from '@vue/test-utils'
import PokemonOptions from '@/components/PokemonOptions'
import { pokemonsMock } from '../mocks/pokemons.mock'

describe('PokemonOptions Component', () => {
    
    let wrapper

      beforeEach(() => {
        wrapper = shallowMount(PokemonOptions, {
            props: {
                pokemons: pokemonsMock 
            }
        })
      })

    test('Debe de hacer match con el Snapshot', () => {
        expect( wrapper.html() ).toMatchSnapshot()
    })

    test('debe de mostrar las 4 opciones correctamente', () => {

        const liPokemons = wrapper.findAll('li')
        const pokemons = liPokemons.map( ( pokemon, index ) => 
        (
            {
                name: pokemon.text(),
                id: index + 1
            }
        ))
        
        expect( pokemons.length ).toBe(4)
        expect( pokemons ).toEqual( pokemonsMock )

    })

    test('Debe de emitir "selectedPokemon" con sus respectivos parametros al hacer click', () => {

        const [li1, li2, li3, li4] = wrapper.findAll('li')

        li1.trigger('click')
        li2.trigger('click')
        li3.trigger('click')
        li4.trigger('click')

        expect( wrapper.emitted('selectedPokemon').length ).toBe(4)
        expect( wrapper.emitted('selectedPokemon')[0] ).toEqual([1])
        expect( wrapper.emitted('selectedPokemon')[1] ).toEqual([2])
        expect( wrapper.emitted('selectedPokemon')[2] ).toEqual([3])
        expect( wrapper.emitted('selectedPokemon')[3] ).toEqual([4])
    })

})