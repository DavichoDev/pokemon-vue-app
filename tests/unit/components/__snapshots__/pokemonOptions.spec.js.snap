// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions Component Debe de hacer match con el Snapshot 1`] = `
<div class="options-container fade-in">
  <ul>
    <li>bulbasaur</li>
    <li>ivysaur</li>
    <li>venusaur</li>
    <li>charmander</li>
  </ul>
</div>
`;
